server/odc-service/src/main/java/com/oceanbase/odc/service/session/ConnectSessionService.java

```diff
@@ -276,7 +276,8 @@ public ConnectionSession create(@NotNull CreateSessionReq req) {
         DefaultConnectSessionIdGenerator idGenerator = new DefaultConnectSessionIdGenerator();
         idGenerator.setDatabaseId(req.getDbId());
         idGenerator.setFixRealId(StringUtils.isBlank(req.getRealId()) ? null : req.getRealId());
-        idGenerator.setHost(stateHostGenerator.getHost());
+        String host = cloudMetadataClient.supportsCloudMetadata() ? req.getFrom() : stateHostGenerator.getHost();
```

Comment:
`req.getFrom` may be null and we should consider of this:
```java
if (!cloudMetadataClient.supportsCloudMetadata() || req.getFrom() == null) {
    idGenerator.setHost(stateHostGenerator.getHost());
} else {
     idGenerator.setHost(req.getFrom());
}
```

Reply:
ok